.circleci/config.yml

```diff
@@ -29,7 +29,7 @@ jobs:
           export WEBAPP_GIT_COMMIT=$(git rev-parse HEAD)
           echo "$WEBAPP_GIT_COMMIT"
 
-          trap 'npm ci && make build' ERR
+          trap 'export WEBAPP_GIT_COMMIT=$(git rev-parse HEAD^); echo "previous $WEBAPP_GIT_COMMIT"; curl -f -o ./dist.tar.gz https://pr-builds.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz; mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1;' ERR
```

Comment:
I wonder if this is the right fix. Building with the previous commit will result in incorrect builds where server is a commit x and webapp is at commit x-1.

IMO, I don't think the issue is with `npm ci && make build`. The problem is with the `trap` itself. When the error happens, the trap is executed and `make build` actually happens - as you can see [here](https://app.circleci.com/pipelines/github/mattermost/mattermost-server/27466/workflows/01fbc8ff-aea4-4f6c-ac80-0dc08ac3d0a8/jobs/192415).

But eventually, at the end it errors out with:
```
Exited with code exit status 22
CircleCI received exit code 22
```

Perhaps we should add a loop to check existence of the latest build for 5 times, and if not, then move forward with the manual build.

Reply:
So strange. OK, updated.